### PR TITLE
OH2 180 update age field computation

### PR DESCRIPTION
**Pull request by SilverD3**

### CODE REVIEW

File: src/components/accessories/patientDataForm/PatientDataForm.tsx [hunk 1]
```diff
@@ -98,6 +99,50 @@ const PatientDataForm: FunctionComponent<TProps> = ({
     }))
   );
 
+  const allAgeTypes = useSelector(
+    (state: IState) => state.ageTypes.getAllAgeTypes.data
+  );
+
+  const getBirthDateAndAge = (
```
Comment (axelpezzo) [inline src/components/accessories/patientDataForm/PatientDataForm.tsx:106]: Can we have an hook for this?

File: src/components/accessories/patientDataForm/PatientDataForm.tsx [hunk 2]
```diff
@@ -98,6 +99,50 @@ const PatientDataForm: FunctionComponent<TProps> = ({
...
+    ageType: string,
```
Comment (axelpezzo) [inline src/components/accessories/patientDataForm/PatientDataForm.tsx:107]: Can we have a enum or a deterministic string?

File: src/components/accessories/patientDataForm/PatientDataForm.tsx [hunk 3]
```diff
@@ -98,6 +99,50 @@ const PatientDataForm: FunctionComponent<TProps> = ({
...
+    formattedValues: Record<string, any>
+  ): { birthDate: string; age: number } => {
+    if (ageType == "agetype") {
```
Comment (axelpezzo) [inline src/components/accessories/patientDataForm/PatientDataForm.tsx:110]: I think that a switch case would be more appropriate for the code reading in this case

File: src/components/accessories/patientDataForm/PatientDataForm.tsx (hunk 1 above)
Comment (SilverD3) [reply src/components/accessories/patientDataForm/PatientDataForm.tsx:106]: I'm not getting you well. Do you suggest to create an hook like the existing one `useCityOptions` ?

Comment (SilverD3) [reply src/components/accessories/patientDataForm/PatientDataForm.tsx:106]: Also, the function `getBirthDateAndAge()` is only used in `PatientDataForm` Component.

Comment (axelpezzo) [inline src/components/accessories/patientDataForm/PatientDataForm.tsx:106]: Sorry, I try to explain my self properly: can we move this method under the libraries folder? Like under formDataHandling that seems to me the right place to store this function.

File: src/components/accessories/patientDataForm/PatientDataForm.tsx (hunk 3 above)
Comment (SilverD3) [reply src/components/accessories/patientDataForm/PatientDataForm.tsx]: Okay

File: src/libraries/formDataHandling/functions.ts [hunk 4]
```diff
@@ -247,3 +249,64 @@ export const isFieldSuggested = (
 ) => {
   return formCustomization.suggestedFields.includes(fieldName);
 };
+
+export const getBirthDateAndAge = (
+  ageType: string,
```
Comment (axelpezzo) [inline src/libraries/formDataHandling/functions.ts:254]: Can we provide an enum for this field?

**Review by axelpezzo — APPROVED** [review]

**Pull request merged**